### PR TITLE
Update RPC handler role/usage (RIPD-557):

### PR DESCRIPTION
**Pull request by bachase**
* Properly use the RPC method to determine required role for HTTP/S RPC calls.
* Charge for malformed RPC calls over HTTP/S

Already reviewed by @vinniefalco and @miguelportilla

### CODE REVIEW

**Comment by bachase** [comment]
Rebased on 0.50.0-b2

**Comment by vinniefalco** [comment]
Nice bump in coverage!

File: src/ripple/rpc/impl/ServerHandlerImp.cpp [hunk 1]
```diff
@@ -550,26 +553,13 @@ ServerHandlerImp::processRequest (Port const& port,
...
     }
 
-    // Parse id now so errors from here on will have the id
-    //
-    // VFALCO NOTE Except that "id" isn't included in the following errors.
-    //
-    Json::Value const& id = jsonRPC [jss::id];
+    /* ---------------------------------------------------------------------- */
+    // Determine role/usage so we can charge for invalid requests
+    Json::Value const& id = jsonRPC ["id"];
```
Comment (HowardHinnant) [inline src/ripple/rpc/impl/ServerHandlerImp.cpp]: Can't this line just be removed now?  And if not, why use `"id"` in place of `jss::id`?

Comment (bachase) [inline src/ripple/rpc/impl/ServerHandlerImp.cpp]: Good catch!  I added it accidentally when cleaning up the rebase.

**Comment by nbougalis** [comment]
Merged as 3c4d3b10c1299607e7d889e83fe74ed5d7425d48.

**Pull request closed without merging**